### DISCUSSION TITLE
USB device issues.

### DISCUSSION THREAD

**Discussion by dshuman52** [procount/pinn#853] [Q&A]
My 64GB USB2.0 device seems to have a quirk; I may have chosen poorly but the form factor was ideal for the raspi 3 (that only has USB2.0 ports) I intended to install it on.  Under the stock raspberry pi os a error message appears that indicated the need for dosfstools and mtools.  When I tried to install both dosfstools was already installed but mtools was not.  I found [https://github.com/procount/pinn/wiki/How-to-Create-a-Multi-Boot-SD-card-out-of-2-existing-OSes-using-PINN](url) and [https://github.com/procount/pinn/blob/master/README_PINN.md#how-to-create-a-custom-os-version](url) that I can create a custom os to load into pinn with mtools included.  My question is whether or not PINN needs to have mtools installed.  I have a feeling because it reads fat32 that dosfstools is already installed there too.  Can I add mtools to PINN myself?

**Comment by procount** [comment]
PINN already has dosfstools and mtools installed.
I would be surprised if raspberry Pi OS didn't have them installed already.
Creating a Custom OS version is a lot of effort just for including mtools. It's much easier to just install it afterwards.
If you must create one (just for your own purposes) I suggest installing Raspios, adding mtools, then using PINN to back it up. You effectively end up with a custom OS, although it includes your initial configuration of Username & password etc.

**Comment by dshuman52** [comment]
Installing it afterwards is not possible when the drive fails to complete the first boot process.  That is the quirk -- i/o problems before the first boot completes.  I could not even format the usb devices until I installed mtools on the base RasPI/Debian  release I was using on the Ras Pi -- because it was a Ras PI 3 the OS I was using was probably bullseye instead of bookworm  Intended for use with my grandson's programming "tinker toys".
Thanks for the prompt response.
    On Friday, December 6, 2024 at 11:27:24 AM EST, procount ***@***.***> wrote:  
 
 


PINN already has dosfstools and mtools installed.
I would be surprised if raspberry Pi OS didn't have them installed already.
Creating a Custom OS version is a lot of effort just for including mtools. It's much easier to just install it afterwards.
If you must create one (just for your own purposes) I suggest installing Raspios, adding mtools, then using PINN to back it up. You effectively end up with a custom OS, although it includes your initial configuration of Username & password etc.
